core/templates/dev/head/domain/skill/MisconceptionObjectFactory.ts

```diff
@@ -17,73 +17,71 @@
  * misconceptions.
  */
 
-var oppia = require('AppInit.ts').module;
-
-oppia.factory('MisconceptionObjectFactory', [
-  function() {
-    var Misconception = function(id, name, notes, feedback) {
-      this._id = id;
-      this._name = name;
-      this._notes = notes;
-      this._feedback = feedback;
-    };
-
-    Misconception.prototype.toBackendDict = function() {
-      return {
-        id: this._id,
-        name: this._name,
-        notes: this._notes,
-        feedback: this._feedback
-      };
-    };
-
-    // TODO (ankita240796) Remove the bracket notation once Angular2 gets in.
-    /* eslint-disable dot-notation */
-    Misconception['createFromBackendDict'] = function(
-    /* eslint-enable dot-notation */
-        misconceptionBackendDict) {
-      return new Misconception(
-        misconceptionBackendDict.id,
-        misconceptionBackendDict.name,
-        misconceptionBackendDict.notes,
-        misconceptionBackendDict.feedback);
-    };
-
-    // TODO (ankita240796) Remove the bracket notation once Angular2 gets in.
-    /* eslint-disable dot-notation */
-    Misconception['create'] = function(id, name, notes, feedback) {
-    /* eslint-enable dot-notation */
-      return new Misconception(id, name, notes, feedback);
-    };
-
-    Misconception.prototype.getId = function() {
-      return this._id;
-    };
-
-    Misconception.prototype.getName = function() {
-      return this._name;
-    };
-
-    Misconception.prototype.setName = function(newName) {
-      this._name = newName;
-    };
-
-    Misconception.prototype.getNotes = function() {
-      return this._notes;
-    };
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
 
-    Misconception.prototype.setNotes = function(newNotes) {
-      this._notes = newNotes;
+export class Misconception {
+  _id: string;
```

Comment:
Verify the type here. Also, are we expecting any test to fail if there isn't any correct type?

Reply:
Verified with the corresponding spec.
There would be a TS typeError.